### PR TITLE
test: use oslo instead of manual btoa

### PR DESCRIPTION
**Pull request by danielgent**
This is probably enough for a first round... ~I'll do another PR where I find&replace more native `btoa` calls with Oslo helpers~ I'm not sure if I should actually as those calls aren't base64 encoding URLs

### CODE REVIEW

File: test/utils/parse-jwt.spec.ts [hunk 1]
```diff
@@ -22,14 +24,16 @@ describe("parseJwt", () => {
...
-    expect(parseJwt(tokenWithSpecialChars)).toEqual(
-      expectedPayloadWithSpecialChars,
-    );
+    const decodedToken = parseJWT(tokenWithSpecialChars);
+
+    expect(decodedToken!.payload).toEqual(expectedPayloadWithSpecialChars);
   });
 
-  it("should throw an error for an invalid token", () => {
+  it("should return null for an invalid token", () => {
```
Comment (danielgent) [inline test/utils/parse-jwt.spec.ts:32]: :eyes: is this a breaking change?

File: test/utils/stateEncode.spec.ts [hunk 2]
```diff
@@ -12,15 +12,15 @@ describe("stateEncode", () => {
       user: "user",
     };
     expect(stateEncode(state)).toEqual(
-      "eyJ1c2VySWQiOiIxMjMiLCJhdXRoUGFyYW1zIjoiYWJjIiwibm9uY2UiOiJ4eXoiLCJzdGF0ZSI6InN0YXRlIiwic2lkIjoic2lkIiwidXNlciI6InVzZXIifQ",
+      "eyJ1c2VySWQiOiIxMjMiLCJhdXRoUGFyYW1zIjoiYWJjIiwibm9uY2UiOiJ4eXoiLCJzdGF0ZSI6InN0YXRlIiwic2lkIjoic2lkIiwidXNlciI6InVzZXIifQ==",
```
Comment (danielgent) [inline test/utils/stateEncode.spec.ts:15]: this is more correct/safer with the padding :+1:

File: src/utils/parse-jwt.ts [hunk 3]
```diff
@@ -1,38 +1,11 @@
...
-  const decoder = new TextDecoder("utf-8");
-
-  return decoder.decode(utf8Bytes);
-}
-
-export function encodeToBase64(text: string): string {
-  // Use TextEncoder to encode the input string to UTF-8 bytes
-  const encoder = new TextEncoder();
-  const utf8Bytes = encoder.encode(text);
+export function parseJwt(token: string): any {
```
Comment (danielgent) [inline src/utils/parse-jwt.ts:3]: I'm keeping this function to maintain compatibility *and* not have a PR with a hundred changed files :smile:

File: test/utils/stateEncode.spec.ts (hunk 2 above)
Comment (markusahlstrand) [reply test/utils/stateEncode.spec.ts:15]: Interesting.. Wouldn't have expected that. Hope it won't be an issue

**Review by markusahlstrand — APPROVED** [review]
LGTM

File: test/utils/stateEncode.spec.ts (hunk 2 above)
Comment (danielgent) [reply test/utils/stateEncode.spec.ts:15]: :see_no_evil: maybe at the moment we release right? :smile:  Some poor users will have their flow ruined... maybe

**Pull request merged**